### ISSUE TITLE
Feature Request : ORPO

### ISSUE THREAD

**Issue by nivibilla** [pytorch/torchtune#894]
Hi,

First of all, thank you for this library! Very clean and I appreciate that all I need is pytorch!

I wanted to make an issue for the integration of [ORPO](https://arxiv.org/abs/2403.07691), not needing to do SFT before the RLHF step is huge since it saves a lot of compute when training on preference data. Hoping it can be integrated into torch tune (with Lora support if possible)!

There is an existing [integration](https://github.com/huggingface/trl/blob/main/trl/trainer/orpo_trainer.py) into TRL

Thanks!

**Comment by nivibilla** [comment]
Hi @kartikayk 

Ik not that experienced in writing custom training loops. Mainly a huggingface user haha. I'd be no better than llama 3 70b attempting it 🤣

**Comment by kartikayk** [comment]
I'm partly serious here, but why not train a codellama 70B using torchtune and then see if this gets you the right recipe :)

**Comment by nivibilla** [comment]
@kartikayk finally got round to doing this and saw a new paper called simPO (simple preference optimisation) and it indeed was simpler to implement than orpo. Only real change being the loss function. 

I have a draft pr here #1036, draft since it's quite messy and I feel it has a lot of duplicate code. Also I haven't tested it at all. 

Closing this issue in favour of #1037
